a-novelo/react-certification-2021
# feat[Mini challenge 5]: Routing

## What does this PR do?
- Adds the Login functionality with mocked credentials.
- Adds the Favorites functionality. Videos can be saved when logged in, and displayed in their own page.

## Where should the reviewer start?
- `LoginModal.component.jsx`: Modal to input credentials for login.
- `Favorites.page.jsx`: Page that displays videos that the user has stored in their favorites, only available when logged in.

## How should this be manually tested?
(These steps are to run locally, but you can also access the deploy on Netlify: https://friendly-khorana-bad590.netlify.app/ )

1. Download the branch
2. Run `yarn install`
3. Set a valid YouTube API key in the `.env` file.
4. Run `yarn start`
5. Log in using the mocked credentials: `wizeline` and `Rocks!`
6. Click on a video and add it to favorites; a button has been added below the video description for that, and it should be updated so the same video can be removed now.
7. Access the favorites page, there's a link on the hamburger menu that's on the header.
8. Log out by clicking on the Wizeline icon in the header.

## What are the relevant tickets?
https://gist.github.com/jparciga/2c2375835ea5e26ba3bc2bd6a30f0ee5

## Screenshots
Logged in:
![image](https://user-images.githubusercontent.com/33430718/112417915-21140f80-8cee-11eb-80ed-333f0ce2efd1.png)

Button to add to favorites:
![image](https://user-images.githubusercontent.com/33430718/112417936-2b360e00-8cee-11eb-9d26-338f251bcc8f.png)

Favorites page:
![image](https://user-images.githubusercontent.com/33430718/112417973-38eb9380-8cee-11eb-9bff-f8bf8522ec89.png)

#### Checklist

#### Code Coverage Reporte
![Stmts](https://user-images.githubusercontent.com/33430718/112418000-46a11900-8cee-11eb-899d-a3126e66bf85.png)
